### PR TITLE
Fix gcc warnings

### PR DESCRIPTION
**Pull request by zzam**
Fix a set of random gcc warnings

### CODE REVIEW

File: src/sound/sound_server.cpp [hunk 1]
```diff
@@ -229,10 +231,18 @@ static bool External_Volume(int volume, int oldVolume) {
...
+static bool External_Play(const std::string &file) {
+	return false;
+}
+static bool External_IsPlaying() {
+	return false;
+}
+static bool External_Stop() {
+	return false;
+}
+static bool External_Volume(int volume, int oldVolume) {
```
Comment (Jarod42) [inline src/sound/sound_server.cpp:243]:
```suggestion
static bool External_Volume(int /*volume*/, int /*oldVolume*/) {
```
Else, might produce warning for unused parameters.

File: src/ui/script_ui.cpp [hunk 2]
```diff
@@ -1138,9 +1138,7 @@ static int CclCopyButtonsForUnitType(lua_State *l)
 
 	// Slot identifier
 	const std::string_view fromName = LuaToString(l, 1);
-	CUnitType &from = UnitTypeByIdent(fromName);
 	const std::string_view toName = LuaToString(l, 2);
-	CUnitType &to = UnitTypeByIdent(toName);
```
Comment (Jarod42) [inline src/ui/script_ui.cpp:1142]: Those lines also check types are correct, so
```
    const std::string_view fromName = LuaToString(l, 1);
	const std::string_view toName = LuaToString(l, 2);
	// Check types exist
	UnitTypeByIdent(fromName);
	UnitTypeByIdent(toName);
```
?

File: src/video/shaders.cpp [hunk 3]
```diff
@@ -257,7 +257,7 @@ static GLint FrameCount;
 static GLint OutputSize;
 static GLint TextureSize;
 static GLint InputSize;
-static GLint VertexCoord;
+// static GLint VertexCoord;
```
Comment (Jarod42) [inline src/video/shaders.cpp:260]:
```suggestion
```

File: src/animation/animation_rotate.cpp [hunk 4]
```diff
@@ -69,6 +69,8 @@ void CAnimation_Rotate::Action(CUnit &unit, int & /*move*/, int /*scale*/) const
 			}
 		} else if (unit.CurrentOrder()->Action == UnitAction::Build) {
 			target = static_cast<const COrder_Build *>(order)->GetBuildingUnit();
+		} else {
+			return;
```
[the remark this answers is not in the document]
Comment (Jarod42) [reply src/animation/animation_rotate.cpp:73]: I don't think so.

File: src/video/shaders.cpp (hunk 3 above)
Comment (zzam) [reply src/video/shaders.cpp:260]: fixed

File: src/ui/script_ui.cpp (hunk 2 above)
Comment (zzam) [reply src/ui/script_ui.cpp:1142]: fixed

File: src/sound/sound_server.cpp (hunk 1 above)
Comment (zzam) [reply src/sound/sound_server.cpp:243]: fixed

**Comment by zzam** [comment]
Github does not show my fix-commit in this PR :-(

File: src/video/shaders.cpp (hunk 3 above)
Comment (Jarod42) [reply src/video/shaders.cpp:260]: Seems you forget to push the changes ;-)

Comment (zzam) [reply src/video/shaders.cpp:260]: The commit is on the branch I created the PR from. But github PR does not show the added changes.
Branch with 3 commits: https://github.com/zzam/stratagus/tree/fix-warnings

**Pull request merged**